diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,260 @@
+        if (isFirstComment(node)) {
+            if (node.treeParent.elementType == BLOCK || node.treeParent.elementType == CLASS_BODY)
+                checkFirstCommentSpaces(node)
+            else
+                checkFirstCommentSpaces(node.treeParent) // TreeParent is property. Then check white spaces before property
+            return
+        }
+
+        if (!node.treePrev.isWhiteSpace()) {
+            WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeParent) // If treeParent is property
+            }
+        } else {
+            if (node.treePrev.numNewLines() == 1 || node.treePrev.numNewLines() > 2) {
+                WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    (node.treePrev as LeafPsiElement).replaceWithText("\n\n")
+                }
+            }
+        }
+    }
+
+    private fun checkSpaceBetweenPropertyAndComment(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        if (node.treeParent.elementType == PROPERTY
+                && node.treeParent.firstChildNode != node) {
+            if (!node.treePrev.isWhiteSpace()) {
+                COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    node.treeParent.addChild(PsiWhiteSpaceImpl(" ".repeat(configuration.maxSpaces)), node)
+                }
+            } else if (node.treePrev.text.length != configuration.maxSpaces) {
+                COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    (node.treePrev as LeafPsiElement).replaceWithText(" ".repeat(configuration.maxSpaces))
+                }
+            }
+        }
+    }
+
+    private fun checkWhiteSpaceBeforeComment(node: ASTNode) {
+        if (node.text.trim('/', '*').takeWhile { it == ' ' }.length == APPROPRIATE_COMMENT_SPACES)
+            return
+
+        COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+            val commentText = node.text.drop(2).trim()
+
+            if (node.text.startsWith("//"))
+                (node as LeafPsiElement).replaceWithText("// $commentText")
+            else if(node.text.startsWith("/*"))
+                (node as LeafPsiElement).replaceWithText("/* $commentText")
+        }
+    }
+
+    private fun checkClassComment(node: ASTNode) {
+        if (isFirstComment(node)) {
+            if (node.treeParent.elementType == BLOCK || node.treeParent.elementType == CLASS_BODY)
+                checkFirstCommentSpaces(node)
+            else
+                checkFirstCommentSpaces(node.treeParent)
+
+            return
+        }
+
+        if (!node.treeParent.treePrev.isWhiteSpace() && node.treeParent.elementType != FILE) {
```

Comment:
I think that if `node.treeParent.elementType == FILE` the first condition will throw NPE when trying to access `treePrev`